weldx/tests/asdf_tests/test_asdf_core.py

```diff
@@ -403,13 +403,52 @@ def test_coordinate_system_manager_time_dependencies(
     csm_root.merge(csm_sub_2)
 
     tree = {"cs_hierarchy": csm_root}
-    data = _write_read_buffer(
+    data = write_read_buffer(
         tree, open_kwargs={"copy_arrays": copy_arrays, "lazy_load": lazy_load}
     )
     csm_file = data["cs_hierarchy"]
     assert csm_root == csm_file
 
 
+@pytest.mark.parametrize("copy_arrays", [True, False])
+@pytest.mark.parametrize("lazy_load", [True, False])
+def test_coordinate_system_manager_with_data(copy_arrays, lazy_load):
+    """Test if data attached to a CSM is stored and read correctly."""
+    csm = tf.CoordinateSystemManager("root", "csm")
+    csm.create_cs("cs_1", "root", coordinates=[1, 1, 1])
+    csm.create_cs("cs_2", "root", coordinates=[-1, -1, -1])
+    csm.create_cs("cs_11", "cs_1", coordinates=[1, 1, 1])
+
+    data_11 = SpatialData(coordinates=np.array([[1.0, 2.0, 3.0], [3.0, 2.0, 1.0]]))
+    data_2 = SpatialData(
+        coordinates=np.array(
+            [
+                [0.0, 0.0, 0.0],
+                [1.0, 0.0, 0.0],
+                [1.0, 1.0, 0.0],
+                [0.0, 1.0, 0.0],
+            ]
+        ),
+        triangles=np.array([[0, 1, 2], [0, 2, 3]], dtype="uint32"),
+    )
+
+    csm.assign_data(data_11, "data_11", "cs_11")
+    csm.assign_data(data_2, "data_2", "cs_2")
+
+    tree = {"csm": csm}
+    buffer = write_read_buffer(
+        tree, open_kwargs={"copy_arrays": copy_arrays, "lazy_load": lazy_load}
+    )
+    csm_buffer = buffer["csm"]
+
+    for data_name in csm.data_names:
+        sd = csm.get_data(data_name)
+        sd_buffer = csm_buffer.get_data(data_name)
+        assert np.allclose(sd.coordinates.data, sd_buffer.coordinates.data)
+        if sd.triangles is not None:
+            assert np.allclose(np.array(sd.triangles), np.array(sd_buffer.triangles))
```

Comment:
use `==` here once implemented

Reply:
Doesn't compare equal:

~~~ 
SpatialData(coordinates=<xarray.DataArray 'core/ndarray' (n: 2, c: 3)>
vs.
SpatialData(coordinates=<xarray.DataArray (n: 2, c: 3)>
~~~

The data seems to be equal, but since compare nested is using xarrays `identical` function, we get a `False` returned.
I remember that we had trouble with the `identical` function before, since it also considers the order of dimensions, coordinates etc. Alternatively, we can use `equals` for a more relaxed comparison but I am not sure if this is really what we want.

Reply:
I think this happens because of the `.name` property that get's created when reading arrays from asdf files (due to `NDArrayType ` wrapper)

see #110 and a possible workaround (reading explicitly as `np.asarray` from tree here https://github.com/BAMWelDX/weldx/commit/1dd1f23ef76c59cbc8c198d00bdf97d049e3c3f7)

maybe we should ask for a solution over at asdf

Reply:
I'll try the workaround for now.

Reply:
Works now. Think we can merge